cucumber-jparest/src/main/java/uk/gov/homeoffice/digital/sas/cucumberjparest/JpaTestContext.java

```diff
@@ -0,0 +1,113 @@
+package uk.gov.homeoffice.digital.sas.cucumberjparest;
+
+import static java.util.Map.entry;
+import static uk.gov.homeoffice.digital.sas.cucumberjparest.ServiceRegistry.SERVICE_REGISTRY_SYSTEM_PROPERTY_NAME;
+
+import io.cucumber.spring.ScenarioScope;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import uk.gov.homeoffice.digital.sas.cucumberjparest.utils.SerialisationUtil;
+
+/**
+ * Class used by the {Link ContextConfiguration} annotation to configure an
+ * {@link org.springframework.context.ApplicationContext ApplicationContext} for integration tests.
+ * In this case for the cucumber tests.
+ */
+@Configuration
+public class JpaTestContext {
+
+  public static final Map<String, Class<?>> classSimpleStrings = Map.ofEntries(
+      entry("String", String.class),
+      entry("Object", Object.class),
+      entry("Integer", Integer.class),
+      entry("Boolean", Boolean.class),
+      entry("Decimal", Double.class),
+      entry("Map", Map.class),
+      entry("List", List.class),
+      entry("Instant", Instant.class));
+  private static final String SERVICE_REGISTRY_SYSTEM_PROPERTY_EL
+      = "#{systemProperties['" + SERVICE_REGISTRY_SYSTEM_PROPERTY_NAME + "']}";
+
+  @Value(SERVICE_REGISTRY_SYSTEM_PROPERTY_EL)
+  private String serviceRegistryString;
+
+  /**
+   * The service registry needs to be accessed to by the Cucumber context and the test runner so
+   * that the test runner can spin up an api and pass its address to the service registry.
+   */
+  @Bean
+  public ServiceRegistry serviceRegistry() {
+    Map<String, String> servicesMap = SerialisationUtil.stringToMap(serviceRegistryString);
+    return new ServiceRegistry(servicesMap);
+  }
+
+  /**
+   * PersonaManager per scenario.
+   *
+   * @return PersonaManager
+   */
+  @ScenarioScope
+  @Bean
+  public PersonaManager personaManager() {
+    return new PersonaManager();
+  }
+
+  /**
+   * HttpResponseManager per scenario.
+   *
+   * @return HttpResponseManager
+   */
+  @ScenarioScope
+  @Bean
+  public HttpResponseManager httpResponseManager() {
+    return new HttpResponseManager();
+  }
+
+  /**
+   * ScenarioState per scenario.
+   *
+   * @return ScenarioState
+   */
+  @ScenarioScope
+  @Bean
+  public ScenarioState scenarioState() {
+    return new ScenarioState();
+  }
+
+  /**
+   * PayloadManager per scenario.
+   *
+   * @return PayloadManager
+   */
+  @ScenarioScope
+  @Bean
+  public PayloadManager payloadManager() {
+    return new PayloadManager();
+  }
+
+  /**
+   * Singleton JpaRestApiClient.
+   *
+   * @return JpaRestApiClient
+   */
+  @Bean
+  public JpaRestApiClient jpaRestApiClient() {
```

Comment:
should this be
```
public JpaRestApiClient jpaRestApiClient(ServiceRegistry serviceRegistry) {
  return new JpaRestApiClient(serviceRegistry);
}
```

Reply:
I think both are valid approaches. The one used here is as per the example provided in [Spring Reference Guide](https://docs.spring.io/spring-javaconfig/docs/1.0.0.M4/reference/html/ch02s02.html#injecting-dependencies)

Reply:
Wrote a comment on this but didn't realised it doesn't get added when you click resolve conversation.
I'm happy to leave as is but was just adding info on why I prefer the syntax in my original comment.
I like the signature highlighting the dependency but beyond that it's also more extendible if you want to inject the dependency i.e. configure service registry in another config class. However at that point you'll probably need to add a qualifier too and we don't really need that.
I was thinking of this example
https://dzone.com/articles/spring-configuration-and